### DISCUSSION TITLE
Correcting tracking errors from an external tracker

### DISCUSSION THREAD

**Discussion by mm809** [SchmollerLab/Cell_ACDC#188] [Q&A]
Hello, 
Thanks for sharing this neat GUI! Would it be possible to use it for correcting tracking errors from an external tracker? It seems that it might be doable if one converts lineage information into the masks and .h5 files in the same format as the segmentation GUI output. Thanks!

**Comment by ElpadoCan** [comment]
Hi @mm809, I'm happy to hear that you want to try Cell-ACDC! 

It's certainly possible to load and correct the result of an external tracker. It's also possible to add external trackers directly into Cell-ACDC. Which tracker did you use? 

Anyway, to load the result of an external tracker, you will need to convert it to a format compatible with Cell-ACDC. What is the output of this tracker? Do you have the segmentation masks?

The segmentation mask for Cell-ACDC is a NumPy file saved as `.npz`. In Cell-ACDC we have conversion utilities (from .tif for example) but I need a little bit more information about the output of this external tracker. Maybe you could also share the data with me? You can send it to may email address (or Google Drive) padovaf@tcd.ie

Cheers,
Francesco

**Comment by ElpadoCan** [comment]
Hi @mm809,

Thank you for sending me your data, I released `v1.2.4rc52` where I managed to modify module 0. a little to add the possibility to convert also the masks files. 

To update to the latest version, please activate the acdc environment and the run the command `pip install --upgrade cellacdc`.

Next, run module 0. and instead of Bio-Formats choose "Re-structure files". Go through the pop-ups and you should get the Position folders that you can load into Cell-ACDC. 

I attached a screenshot of the parameters for the conversion that I used with your test data.

Let me know if it's clear enough and if it works, thanks! 

Cheers,
Francesco

![conversion_param](https://user-images.githubusercontent.com/55240925/201963904-55d0ea54-7de8-46f6-ad17-93c6e4c1b9e0.png)

**Reply by ElpadoCan** [reply]
I just realised now that I forgot to also look into the tracking info in the mat file. I will do that soon sorry about that

**Reply by mm809** [reply]
Hi Francesco, The mask conversion works perfectly. Thanks!

**Comment by ElpadoCan** [comment]
Hi @mm809 

I just released `v1.2.4rc53` with the new utility we discussed. To update, please activate the `acdc` environment and then run the command `pip install --upgrade cellacdc`.

You can run the new utility for the menu on the top menubar of the Cell-ACDC launcher called `Utilities --> Tracking --> Apply tracking info from tabular data...`. 

Next, go through the wizard and select a Position folder (one with already the segmentation mask in Cell-ACDC format). Hopefully, the process will be self-explanatory.

I attached here a screenshot of the main window where you select which column is what with the selection I used. 

Let me know if it works or if you have issues, thanks! 

Cheers,
Francesco

PS: For the correction of the mother-daughter relationships I think we will need to discuss a bit what you want to achieve because I think it will need some fine-tuning. Nobody started from pre-annotated data so far. We can do that in another issue. Feel free to open it when you are ready.

![image](https://user-images.githubusercontent.com/55240925/203619051-dc9f8a18-3e30-4249-978b-396367766e8d.png)

**Reply by mm809** [reply]
Hi Francesco,

Thanks for the new utility! The conversion works beautifully:-). The converted tracking information also shows up nicely in the 'Viewer' mode. In the 'Segmentation and Tracking' mode, however, tracking seems to be re-done - i.e. in the example file, track 16 goes from frame 18 to frame 61, whereas in the 'Segmentation and Tracking' mode, the same track (given an ID of 35) goes from frame 18 to 60, and was given a new ID at frame 61. I guess acdc is automatically correcting tracking? Could this feature be toggle on/off?

Thank you very much!

Mingwei

PS: Yes I do have a couple of questions on the genealogy - will open an issue on this.

**Reply by ElpadoCan** [reply]
Hi Min, happy to hear that it worked!

Oh yes, I forgot to mention that in your case you should turn off real time tracking. You can turn it off from the top toolbar on the right there is a checkbox called "Disable tracking". Let me know if it works thanks!

And yes you are right, Cell-ACDC has built in real time tracking, but it's based on fast trackers (since it's performed every time you visit a new frame) which needs to sacrifice some complexity (hence accuracy) to be as fast as possible. The two real time trackers that we have so far are YeaZ and a custom one, and both were tuned for tracking budding yeast. The other two trackers (bayesian tracker and trackpy) are more generalist but also way more complex. 

Cheers
Francesco
